corehq/form_processor/backends/sql/update_strategy.py

```diff
@@ -72,6 +72,14 @@ def _apply_case_update(self, case_update, xformdoc):
         if self.case.modified_on is None or modified_on > self.case.modified_on:
             self.case.modified_on = modified_on
 
+        # edge case if form updates case before it's been created (or creation form archived)
+        if not self.case.opened_on:
+            self.case.opened_on = case_update.guess_modified_on()
```

Comment:
You could probably reuse the `modified_on` variable from above. Not sure if this does whatever work `guess_modified_on` does twice or if that is cached in some way.